core/src/main/scala/org/apache/spark/sql/SnappyContext.scala

```diff
@@ -546,41 +548,114 @@ class SnappyContext protected[spark](
     }
   }
 
+  /**
+    * Create an index on a table.
+    * @param indexName Index name which goes in the catalog
+    * @param baseTable Fully qualified name of table on which the index is created.
+    * @param indexColumns Columns on which the index has to be created
+    * @param options Options for indexes. For e.g.
+    *                column table index - ("COLOCATE_WITH"->"CUSTOMER").
+    *                row table index - ("INDEX_TYPE"->"GLOBAL HASH") or ("INDEX_TYPE"->"UNIQUE")
+    */
+  def createIndex(indexName: String,
+                  baseTable: String,
+                  indexColumns: java.util.Map[String, Option[SortDirection]],
+                  options: java.util.Map[String, String]): Unit = {
+    createIndex(indexName, baseTable, indexColumns.asScala.toMap, options.asScala.toMap)
+  }
 
```

Comment:
I think its time to enable scalaStyle on core and cluster too so that such issues are caught beforehand.